homeassistant/components/habitica/sensor.py

```diff
@@ -259,14 +261,102 @@ async def async_setup_entry(
     """Set up the habitica sensors."""
 
     coordinator = config_entry.runtime_data
+    ent_reg = er.async_get(hass)
+    entities: list[SensorEntity] = []
+    description: SensorEntityDescription
+    for description in SENSOR_DESCRIPTIONS:
+        if description.key is HabiticaSensorEntity.HEALTH_MAX:
+            if entity_id := ent_reg.async_get_entity_id(
+                SENSOR_DOMAIN,
+                DOMAIN,
+                f"{config_entry.unique_id}_{description.key}",
+            ):
+                entity_entry = ent_reg.async_get(entity_id)
+                assert entity_entry
+                if entity_entry.disabled:
+                    ent_reg.async_remove(entity_id)
+                    async_delete_issue(
+                        hass,
+                        DOMAIN,
+                        f"deprecated_entity_{description.key}",
+                    )
+                else:
+                    entities.append(HabiticaSensor(coordinator, description))
+                    if entity_used_in(hass, entity_id):
+                        async_create_issue(
+                            hass,
+                            DOMAIN,
+                            f"deprecated_entity_{description.key}",
+                            breaks_in_ha_version="2025.8.0",
+                            is_fixable=False,
+                            severity=IssueSeverity.WARNING,
+                            translation_key="deprecated_entity",
+                            translation_placeholders={
+                                "name": str(
+                                    entity_entry.name or entity_entry.original_name
+                                ),
+                                "entity": entity_id,
+                            },
+                        )
+        else:
+            entities.append(HabiticaSensor(coordinator, description))
+
+    for description in TASK_SENSOR_DESCRIPTION:
+        if entity_id := ent_reg.async_get_entity_id(
+            SENSOR_DOMAIN,
+            DOMAIN,
+            f"{config_entry.unique_id}_{description.key}",
+        ):
+            entity_entry = ent_reg.async_get(entity_id)
+            assert entity_entry
+            if entity_entry.disabled:
+                ent_reg.async_remove(entity_id)
+                async_delete_issue(
+                    hass,
+                    DOMAIN,
+                    f"deprecated_task_entity_{description.key}",
+                )
+            else:
+                entities.append(HabiticaTaskSensor(coordinator, description))
+                if description.key in (
```

Comment:
Maybe we can combine these with the ones below and put the breaking version in the description and use it as a variable?